modules/core/src/main/scala/org/tesselation/infrastructure/cluster/storage/AddressStorage.scala

```diff
@@ -1,20 +1,26 @@
 package org.tesselation.infrastructure.cluster.storage
 
 import cats.effect.Async
-
+import cats.syntax.functor._
+import doobie.implicits._
+import doobie.quill.DoobieContext
+import io.getquill.{Literal, PostgresDialect}
 import org.tesselation.domain.cluster.storage.AddressStorage
 import org.tesselation.infrastructure.db.context.AddressDBContext
-import org.tesselation.infrastructure.db.{DoobieTransactor, schema}
+import org.tesselation.infrastructure.db.doobie.DoobieTransactor
 import org.tesselation.schema.address.{Address, Balance}
 
-import doobie.implicits._
-import doobie.quill.DoobieContextBase
-import io.getquill.{Literal, PostgresDialect}
-
 object AddressStorage {
 
+  type Context = DoobieContext.Postgres[Literal] with AddressDBContext[PostgresDialect, Literal]
+
+  def make[F[_]: Async: DoobieTransactor]: F[AddressStorage[F]] =
```

Comment:
Async probably is not needed there (dunno what `transact` needs). Check MonadThrow with Applicative or ApplicativeError or sth like that